astronomer/providers/apache/livy/example_dags/example_livy.py

```diff
@@ -4,37 +4,324 @@
+    """
+    client = boto3.client("emr", **AWS_S3_CREDS)
+    response = client.describe_cluster(
+        ClusterId=str(task_instance.xcom_pull(key="return_value", task_ids=["cluster_creator"])[0])
+    )
+    while (
+        "MasterPublicDnsName" not in response["Cluster"]
+        and response["Cluster"]["Status"]["State"] != "WAITING"
+    ):
+        print("wait for 11 minutes to get the MasterPublicDnsName")
+        time.sleep(660)
+        response = client.describe_cluster(
+            ClusterId=str(task_instance.xcom_pull(key="return_value", task_ids=["cluster_creator"])[0])
+        )
+        print("current response from ams emr: ", response)
+    task_instance.xcom_push(
+        key="cluster_response_master_public_dns", value=response["Cluster"]["MasterPublicDnsName"]
+    )
+    task_instance.xcom_push(
+        key="cluster_response_master_security_group",
+        value=response["Cluster"]["Ec2InstanceAttributes"]["EmrManagedMasterSecurityGroup"],
+    )
+
 
 with DAG(
     dag_id="example_livy_operator",
-    default_args={"args": [10]},
     schedule_interval=None,
     start_date=datetime(2021, 1, 1),
     catchup=False,
     tags=["example", "async", "deferrable", "LivyOperatorAsync"],
 ) as dag:
+    # [START howto_create_key_pair_file]
+    create_key_pair_file = PythonOperator(
+        task_id="create_key_pair_file",
+        python_callable=create_key_pair,
+    )
+    # [END howto_create_key_pair_file]
+
+    # [START howto_operator_emr_create_job_flow]
+    cluster_creator = EmrCreateJobFlowOperator(
+        task_id="cluster_creator",
+        job_flow_overrides=JOB_FLOW_OVERRIDES,
+    )
+    # [END howto_operator_emr_create_job_flow]
+
+    # [START describe_created_cluster]
+    describe_created_cluster = PythonOperator(
+        task_id="describe_created_cluster", python_callable=get_cluster_details
+    )
+    # [END describe_created_cluster]
 
-    # [START create_livy]
+    # [START add_example_pi_file_in_hdfs]
+    ssh_and_copy_pifile_to_hdfs = PythonOperator(
+        task_id="ssh_and_copy_pifile_to_hdfs",
+        python_callable=ssh_and_run_command,
+        op_kwargs={
+            "path_to_pem_file": f"/tmp/{PEM_FILENAME}.pem",
+            "username": "hadoop",
+            "command": COMMAND_TO_CREATE_PI_FILE,
+        },
+    )
+    # [END add_example_pi_file_in_hdfs]
+
+    # [START add_ip_address_for_inbound_rules]
+    get_and_add_ip_address_for_inbound_rules = PythonOperator(
+        task_id="get_and_add_ip_address_for_inbound_rules",
+        python_callable=add_inbound_rule_for_security_group,
+    )
+    # [END add_ip_address_for_inbound_rules]
+
+    # [START create_airflow_connection_for_livy]
+    create_airflow_connection_for_livy = PythonOperator(
+        task_id="create_airflow_connection_for_livy", python_callable=create_airflow_connection
+    )
+    # [END create_airflow_connection_for_livy]
+
+    # [START run_pi_example_with_no_polling_interval]
```

Comment:
nit: can change r`un_pi_example_with_no_polling_interval` to` run_pi_example_without_polling_interval`

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/235/commits/2d39230044199f16b7a022e0f71c62713bf468b8)